### ISSUE TITLE
Feature Request: Internal Combustion Engine

### ISSUE THREAD

**Issue by FudgePucker** [d3athrow/vgstation13#5163]
I came up with an idea for a new atmos based engine using (if it ever gets merged) the 5 pipe layers.
The problem with most atmos based engines (RUST, SME, TEG) is that they require large chambers.
My new engine idea would remove the need for something like that.
![ice](https://cloud.githubusercontent.com/assets/12202731/8393007/fd53476c-1cca-11e5-9231-0566123265e0.png)
The purple pipe is intake 1, the light blue pipe is intake 2, the red pipe is temperature regulator, the dark blue pipe is coolant, and the gray pipe is exhaust.

The engine takes in from the two intake pipes plasma and O2, requiring to be set up like a gas mixer to have the right ratio, it compresses the gas and sparks it causing it to expand and generate power. The spark is just provided by a power cell and is self charging, however, if you are using very cold fuel, the compression and spark won't be enough, that's where the temperature regulator comes into play, by heating up the fuel as it is in the chamber before it ignites, it can raise it to the point of being able to ignite and generate power. This allows you to use cold fuel for storage and power reasons. The engine also gets very hot, and will make the surrounding area hotter and also will start to melt down, ruining the engine and spilling its intakes all over the room, that's where the coolant comes into place, it cools down the engine itself, not the fuel, so that the engine doesn't radiate heat into the surrounding room or melt down. The exhaust pipe is there, well, because the CO2 has to go somewhere, you can set the engine to just output the exhaust directly into the room instead of using the pipe, but that might be a bad idea, also if your exhaust pipeline gets too clogged then the engine will start choking up and melting down.

This engine will be small and will be able to fit into smaller locations that aren't connected to grid. Perfect for taxi station.
Also I don't know if you already realized where this will be taken on an autism level, but they can also be stacked side by side.
By having multiple engines sharing fuel intakes, regulators, coolant, and exhaust, the ICE (Boy isn't that a fucking awesome abbreviation? 'Hey someone go set up the ICE' 'Hey guys I brought the ICE for this party ship') can power the entire station by just having one giant room of engines.

The point I was making with having a temperature regulator and exhaust is that you can use thermal exchangers to heat back up the temperature regulator line and help cool down the exhaust line, meaning it needs less maintenance.

This engine will be simple, space compact, yet require skilled players to set it up. It requires you to set up the ratio yourself for the fuel intake pipes, meaning if your plasma and O2 are at different temps you need to figure out the right ratio for maximizing burn. It would also be useful if you are working on toxins because you can have specific temperatures of gasses by setting up your ratio right and filtering the exhaust.

Its like having a fully functional burn chamber, toxins lab, TEG, all in one nice small package.

TLDR: It will replace the gas turbine generator as a non science fiction engine using science fiction fuel. It will be small enough to fit in small places that you can't make a full engine with. It will have a somewhat low skill floor but a high skill ceiling. It will have more uses outside of just power. It has the coolest abbreviation ever.

**Comment by CptWad** [comment]
>It has the coolest abbreviation ever.

it does
So can it generate power or does the engine just engine around?

**Comment by FudgePucker** [comment]
It does generate power.
The combustion moves a turbine creating power.
It doesn't create as much as say the singularity or SME, but it takes up less space.
Think of it as a pacman that takes in gasses and can be linked up to atmospherics to be fueled automatically.

**Comment by AlphaBuffalo** [comment]
It looks like a smaller teg

**Comment by ghost** [comment]
Refurbish the pre-ZAS turbines in the code to work on ZAS? It's on the heartstation map if I remember right?
